src/components/experimental/Gallery.astro

```diff
@@ -1,47 +1,37 @@
 ---
-import Icon from 'fulldev-ui/components/Icon.astro'
 import Image from 'fulldev-ui/components/Image.astro'
+import Root from '../Root.astro'
 
-const { images, direction } = Astro.props
+const { images, direction = 'horizontal' } = Astro.props
 ---
 
 <script>
-  const carousel = document.querySelector('.gallery .large-carousel')
-  const icon = document.querySelector('.gallery .icon')
-  const smallImages = document.querySelectorAll('.gallery .small-image')
-
-  carousel &&
-    icon?.addEventListener('click', () => {
-      const carouselWidth = (carousel as HTMLElement).offsetWidth
-      const currentScroll = carousel.scrollLeft
-      const maxScroll = carousel.scrollWidth - carouselWidth
-
-      let scrollAmount = currentScroll + carouselWidth
-      if (scrollAmount > maxScroll) {
-        scrollAmount = 0 // Reset to beginning if we're at the end
-      }
-
-      carousel.scrollTo({
-        left: scrollAmount,
-        behavior: 'smooth',
-      })
-    })
-
-  carousel &&
-    smallImages?.forEach((image, i) => {
-      image.addEventListener('click', () => {
-        const imageWidth = (carousel as HTMLElement).offsetWidth
-        carousel.scrollTo({
-          left: i * imageWidth,
-          behavior: 'smooth',
+  document.addEventListener('DOMContentLoaded', () => {
```

Comment:
Have a look at https://docs.astro.build/en/guides/view-transitions/#lifecycle-events instead.

Also see ThemeSelect.astro and Layout.astro for examples, which are based on the examples in de link above